src/api/app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,84 @@
+# TODO: Please overwrite this comment with something explaining the controller target
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: [:show, :update, :destroy]
+  # Pundit authorization policies control
+  after_action :verify_authorized, except: :index
+  after_action :verify_policy_scoped, only: :index
+
+  # GET /announcements
+  def index
+    @announcements = policy_scope(Announcement.all)
+    xml = if @announcements.exists?
+            @announcements.to_xml(Announcement::DEFAULT_RENDER_PARAMS)
+          else
+            '<announcements></announcements>'
+          end
+    render xml: xml
```

Comment:
Then let's use `to_xml`.

Reply:
just wait, we have a lot of views being built with XMLBuilder. There is something that speak against do int model: You are getting the render concern and pushing to the model, however it belongs to the view.

Reply:
```
$ find . -name '*.builder'  | wc -l
60 
```

Reply:
I think here https://github.com/openSUSE/open-build-service/pull/5024 we saw how the GC kicked in and why in that case to use the XMLBuilder was better.. it may or may not apply here, but I think that's relevant to the decision @bgeuken

Reply:
as reference http://blog.codepath.com/2011/05/16/if-youre-using-to_json-youre-doing-it-wrong/ the "to_json" issue applies too to "to_xml"

Reply:
> vpereira 2 hours ago
> $ cd app/view &&  find . -name '*.builder'  | wc -l
> 60

I am aware of that

Reply:
>  vpereira 2 hours ago
> I think here #5024 we saw how the GC kicked in

In the PR you refer to the xml was constructed via string snippets. Usually the to_xml implementations use an XML builder as well ([Builder::XmlMarkup](https://edgeguides.rubyonrails.org/active_support_core_extensions.html#extensions-to-array-conversions-to-xml) in case of Rails). So I doubt the performance difference is huge.

Reply:
>  vpereira 2 hours ago
> There is something that speak against do in the model: You are getting the "render specific type using this layout concern" and pushing to the model, however it really belongs to the view.

Given that we have a RESTful API I would argue that the API should return the XML representation of a resource. Why would that be different from the model / only bound to the view? :-)

Reply:
That said I am aware that changing all our builder views would be a bigger effort and I am going to add the xml views here.
But I disagree that this has to be like that or that this a good way of handling XML representations of objects.